src/clj/tic_tac_toe_clojure/core.clj

```diff
@@ -0,0 +1,74 @@
+(ns tic-tac-toe-clojure.core (:require [clojure.string :as string]))
+(declare start-game)
+(defn -main [& args]
+  (println "Starting game")
+  (start-game))
+
+(defn create-board []
+  (into [] (repeat 9 "_")))
+
+(def numbered-board
+  ["0" "1" "2" "3" "4" "5" "6" "7" "8"])
+
+(defn build-board [board]
+  (with-out-str (println (string/join "\n" (re-seq #".{1,3}" (apply str board))))))
+
+(defn display-board [board]
+  (print (build-board board)))
+
+(defn get-user-symbol []
+  (do (print "Please choose a symbol (X or O): ") (flush) (read-line)))
+
+(defn get-user-position []
+ (Integer/parseInt (do (print "Please choose a position between 0 and 8: ") (flush) (read-line))))
+
+(defn set-position [board position user-symbol]
+  (assoc board position user-symbol))
+
+(defn swap-player [user-symbol]
+  (if (= "X" user-symbol) "O" "X"))
+
+(defn get-rows [board]
+  (partition 3 board))
+
+(defn get-columns [board]
+  (apply map vector (get-rows board)))
+
+(defn get-diagonals [board]
+  [[(nth board 0) (nth board 4) (nth board 8)]
+   [(nth board 2) (nth board 4) (nth board 6)]])
```

Comment:
Could you use a helper for this like:

```clj
(defn- tiles-at [board moves]
   (map (fn [move] (nth board move)) moves)
```

And then use like:

```clj
(defn get-diagonals [board]
   [(tiles-at [0 4 8]),
    (tiles-at [2 4 6])])
```

Or even shorter for the helper:

```clj
(defn- tiles-at [board moves]
   (map (partial nth board) moves))
```